### PR TITLE
EUI-3802-lint-error-fix: enable lint in github actions

### PR DESCRIPTION
**Pull request by rayliangatos**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)
https://tools.hmcts.net/jira/browse/EUI-3802

Notes:
* Enable Lint in Github actions
* Enable trigger CI/CD pipeline when PR->Release
* Fix all Lint errors

### CODE REVIEW

File: projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts [hunk 1]
```diff
@@ -8,16 +8,16 @@ import { Component, Input, OnDestroy, OnInit, Renderer2, ViewEncapsulation } fro
...
-  constructor(private renderer: Renderer2) {}
+  constructor(private readonly renderer: Renderer2) {}
 
-  ngOnInit() {
+  public ngOnInit() {
     this.renderer.addClass(document.body, 'loading-spinner-in-action');
   }
 
-  ngOnDestroy() {
+  public ngOnDestroy() {
```
Comment (phillip-whitaker) [inline projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts:20]: where is the return type?

File: projects/exui-common-lib/src/lib/services/loading/loading.service.ts [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+import { distinctUntilChanged } from 'rxjs/operators';
 
 
 @Injectable({
   providedIn: 'root'
 })
+export abstract class HasLoadingState {
+  public get isLoading(): Observable<boolean> {
+    return;
+  }
```
Comment (phillip-whitaker) [inline projects/exui-common-lib/src/lib/services/loading/loading.service.ts:12]: that isn't returning an observable boolean....

File: projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts [hunk 3]
```diff
@@ -8,16 +8,16 @@ import { Component, Input, OnDestroy, OnInit, Renderer2, ViewEncapsulation } fro
...
-  @Input() zIndex: number = 1000;
-  @Input() loadingText: string = 'Page can take up to 30 seconds to load';
+  @Input() public zIndex: number = 1000;
+  @Input() public loadingText: string = 'Page can take up to 30 seconds to load';
 
-  constructor(private renderer: Renderer2) {}
+  constructor(private readonly renderer: Renderer2) {}
 
-  ngOnInit() {
+  public ngOnInit() {
```
Comment (paul-graham) [inline projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts:16]: If we're going to introduce `public` at this stage, we should also give it a `: void` return type.

File: projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts (hunk 1 above)
Comment (paul-graham) [inline projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts:20]: If we're going to introduce `public` at this stage, we should also give it a `: void` return type.

**Review by paul-graham — CHANGES_REQUESTED** [review]
Minor changes requested for lifecycle return types.

**Review by leepavlouhmcts — APPROVED** [review]
as long as the comments are resolved, i approve

File: projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts (hunk 1 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts]: done

File: projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts (hunk 3 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts]: done

File: projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts (hunk 1 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/loading-spinner/loading-spinner.component.ts]: done

**Review by phillip-whitaker — APPROVED** [review]
Review comments addressed

File: projects/exui-common-lib/src/lib/services/loading/loading.service.ts (hunk 2 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/services/loading/loading.service.ts]: this is exisiting code I just fixed the lint issue, but anyway I have now changed it to abstract method

**Review by paul-graham — APPROVED** [review]
Lovely. Thanks, @rayliangatos. 👍

**Pull request merged**